bionic/provider.py

```diff
@@ -142,7 +152,7 @@ def __init__(self, wrapped_provider, attr_name, attr_value):
 
 
 class ProtocolUpdateProvider(WrappingProvider):
-    def __init__(self, wrapped_provider, protocol=None, protools=None):
+    def __init__(self, wrapped_provider, protocol=None, protocols=None):
```

Comment:
Nice catch!  But it looks like we don't even use this argument.  I think it's vestigial and can be removed.